docs/deprecated.md

```diff
@@ -0,0 +1,39 @@
+# Deprecated functionality
```

Comment:
A link to the new document should be added in master README.md, as it is done for instance in https://github.com/telefonicaid/lightweightm2m-iotagent#usage

Reply:
@fgalan will add the link in master README.md onces deprecated.md gets merged.

Reply:
It is better to do it in this same PR, for the sake of consistency. Or maybe I'm missing some reason which precludes doing so?

Reply:
@fgalan when I will add a link of deprecated file (https://github.com/telefonicaid/iotagent-json/tree/master/docs/deprecated.md). It will not be found because it isn't merged in master yet.

Reply:
@fgalan if I will add the link of deprecated file(https://github.com/telefonicaid/iotagent-json/tree/master/docs/deprecated.md) in this, it won't exist as it isn't merged into master yet.

Reply:
I have added the link in README.md. When deprecate.md will be merged, https://github.com/telefonicaid/iotagent-json/tree/master/docs/deprecated.md will be existent in README.md usage module. Thank you!